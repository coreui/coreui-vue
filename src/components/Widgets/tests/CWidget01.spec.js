import { shallowMount } from '@vue/test-utils'
import Component from '../CWidget01'

const ComponentName = 'CWidget01'
const defaultWrapper = shallowMount(Component)
const customWrapper = shallowMount(Component, {
  context:{
    props: {
      variant: 'info',
      inverse: true,
      value: 30,
      text:'test',
      header: 'test',
      footer: 'test'
    }
  }
})

describe(ComponentName, () => {
  it('has a name', () => {
    expect(Component.name).toMatch(ComponentName)
  })
  it('is functional component', () => {
    expect(defaultWrapper.isFunctionalComponent).toBe(true)
  })
  it('correctly render component with default props', () => {
    expect(defaultWrapper.findAll('div').at(0).attributes().class).toMatch('card')
    expect(defaultWrapper.findAll('div').at(2).text()).toMatch('header:string')
    expect(defaultWrapper.findAll('div').at(3).text()).toMatch('text:string')
    expect(defaultWrapper.find('small').text()).toMatch('footer:string')
    expect(defaultWrapper.find('cprogress-stub').exists()).toBe(true)
    expect(defaultWrapper.find('cprogress-stub').attributes().color).toBe('success')
    expect(defaultWrapper.find('cprogress-stub').attributes().class).toBe('progress-xs my-3 mb-0')
    expect(defaultWrapper.find('cprogress-stub').attributes().value).toBe('25')
  })
  it('matches default props snapshot', () => {
    expect(defaultWrapper.element).toMatchSnapshot()
  })
  it('correctly render component with custom props', () => {
    expect(customWrapper.findAll('div').at(0).attributes().class).toMatch('card text-white bg-info')
    expect(customWrapper.findAll('div').at(2).text()).toMatch('test')
    expect(customWrapper.findAll('div').at(3).text()).toMatch('test')
    expect(customWrapper.find('small').text()).toMatch('test')
    expect(customWrapper.find('cprogress-stub').exists()).toBe(true)
    expect(customWrapper.find('cprogress-stub').attributes().color).toBe('')
    expect(customWrapper.find('cprogress-stub').attributes().class).toBe('progress-xs my-3 mb-0 progress-white')
    expect(customWrapper.find('cprogress-stub').attributes().value).toBe('30')
  })
  it('matches custom props snapshot', () => {
    expect(customWrapper.element).toMatchSnapshot()
  })
})
