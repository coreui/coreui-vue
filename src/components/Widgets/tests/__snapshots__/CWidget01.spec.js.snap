// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CWidget01 matches custom props snapshot 1`] = `
<div
  class="card text-white bg-info"
>
  <div
    class="card-body"
  >
    <div
      class="h4 m-0"
    >
      test
    </div>
     
    <div>
      test
    </div>
     
    <cprogress-stub
      class="progress-xs my-3 mb-0 progress-white"
      color=""
      max="100"
      precision="0"
      value="30"
    />
     
    <small
      class="text-muted"
    >
      test
    </small>
  </div>
</div>
`;

exports[`CWidget01 matches default props snapshot 1`] = `
<div
  class="card"
>
  <div
    class="card-body"
  >
    <div
      class="h4 m-0"
    >
      header:string
    </div>
     
    <div>
      text:string
    </div>
     
    <cprogress-stub
      class="progress-xs my-3 mb-0"
      color="success"
      max="100"
      precision="0"
      value="25"
    />
     
    <small
      class="text-muted"
    >
      footer:string
    </small>
  </div>
</div>
`;
